src/server/routes/api/search.ts

```diff
@@ -0,0 +1,69 @@
+import { Router } from "express";
+import db from "../../db";
+import patterns from "../../db/queries/patterns";
+import { IPatternTable, Tag, Tags } from "../../types";
+
+const router = Router();
+
+//GET /api/search/title/:param
+router.get("/title/:queryString", async (req, res, next) => {
+	let title = req.params.queryString;
+	try {
+		const result = await db.search.findByTitle(title);
+		if (result.length === 0) {
+			return res.status(404).json({ message: "No patterns found" });
+		}
+		res.json({ result, message: "Patterns found" });
+	} catch (error) {
+		console.error("Error fetching patterns:", error);
+		next(error);
+	}
+});
+
+//POST /api/search/tag
+router.post("/tag", async (req, res, next) => {
+	try {
+		const finalPatterns: IPatternTable[] = [];
+		let tags: Tags;
+
+		try {
+			tags = JSON.parse(req.body.tagList);
+		} catch (err) {
+			return res.status(400).json({ message: "Invalid JSON format" });
+		}
+		const idArray: number[] = tags.map((tag: Tag) => tag.id);
+
+		const patternPromises = idArray.map((id: number) => {
+			return db.search.findByTags(id);
+		});
+
+		//Zach what type is this??
+		let result;
```

Comment:
Since it is going to hold the result of all your promises, I believe this should be an array of whatever db.search.findByTags returns!

Reply:
LOL I love that you found my comment asking for your help. I'll dig into this and type it correctly~

Reply:
Yeah I need more help on this. I'm going to resolve it and chat with you about it cause it's a bit more complicated than I can resolve on my own